### PR TITLE
Completing line_search_strategy python definition

### PR DESCRIPTION
**Pull request by loumalouomega**
The Previous definition was leading to an bad signature when invoked from python

### CODE REVIEW

File: kratos/python/add_strategies_to_python.cpp [hunk 1]
```diff
@@ -272,8 +272,12 @@ namespace Kratos
                     ;
 
             class_< LineSearchStrategy< SparseSpaceType, LocalSpaceType, LinearSolverType >, bases< ResidualBasedNewtonRaphsonStrategyType >, boost::noncopyable >
-                    ("LineSearchStrategy",
-                    init < ModelPart&, BaseSchemeType::Pointer, LinearSolverType::Pointer, TConvergenceCriteriaType::Pointer, int, bool, bool, bool >())
+                    ("LineSearchStrategy", init < ModelPart&, BaseSchemeType::Pointer, LinearSolverType::Pointer, TConvergenceCriteriaType::Pointer, int, bool, bool, bool >())
+                    .def(init < ModelPart&, BaseSchemeType::Pointer, LinearSolverType::Pointer, TConvergenceCriteriaType::Pointer, BuilderAndSolverType::Pointer, int, bool, bool, bool >())
```
Comment (jcotela) [inline kratos/python/add_strategies_to_python.cpp:276]: Doesn't this init method take the same arguments as the init<> template parameter in the line above? If so, you are exposing to python the same method twice. You don't need to redefine it here.

**Review by jcotela — CHANGES_REQUESTED** [review]
In general, I dislike exposing all possible combinations of default parameters to python, because it makes it hard to know exactly what parameters you are passing in practice (you need to check the python script, this file and the C header). I'd prefer to have two or three constructors maximum, for example one with all arguments and another with only the essential ones. However, I'm willing to approve if you have a use case.

By the way, if you need this level of flexibility, it is probably cleaner to pass a Kratos::Parameters instance.

File: kratos/python/add_strategies_to_python.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/python/add_strategies_to_python.cpp:276]: It is not the same BuilderAndSolverType::Pointer is missing

**Comment by loumalouomega** [comment]
@jcotela I did the requested changes

**Review by jcotela — APPROVED** [review]

**Pull request merged**